libarchive/archive_read_support_format_raw.c

```diff
@@ -116,11 +116,15 @@ archive_read_format_raw_read_header(struct archive_read *a,
 
 	a->archive.archive_format = ARCHIVE_FORMAT_RAW;
 	a->archive.archive_format_name = "raw";
+
+	/* I'm deliberately leaving most fields unset here. */
 	archive_entry_set_pathname(entry, "data");
 	archive_entry_set_filetype(entry, AE_IFREG);
 	archive_entry_set_perm(entry, 0644);
-	/* I'm deliberately leaving most fields unset here. */
-	return (ARCHIVE_OK);
+	archive_entry_set_size(entry, a->filter->avail);
```

Comment:
It's better to leave the size unset ("unknown") here.  Some downstream consumers get cranky if this is set but wrong.